File: backend/app/schemas/note.py
from pydantic import BaseModel, Field, validator
from typing import Optional
from datetime import datetime

class NoteBase(BaseModel):
    title: str = Field(..., min_length=1)
    content: str = Field(..., min_length=10)

    @validator('title')
    def title_must_not_be_empty(cls, v):
        if not v.strip():
            raise ValueError('Title cannot be empty')
        return v

class NoteCreate(NoteBase):
    pass

class NoteResponse(NoteBase):
    id: int
    sentiment: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

class NoteAnalysis(BaseModel):
    sentiment: str 

File: frontend/app/types/note.ts
export interface Note {
  id: number;
  title: string;
  content: string;
  sentiment: string | null;
  created_at: string;
  updated_at: string;
}

export interface NoteCreate {
  title: string;
  content: string;
}

export interface NoteAnalysis {
  sentiment: string;
}

export type SentimentType = 'positive' | 'neutral' | 'negative'; 